discovery-provider/src/challenges/profile_challenge.py

```diff
@@ -0,0 +1,116 @@
+    # Helpers
+
+    def _handle_profile_updates(self, completion_map, users):
+        for user in users:
+            completion = completion_map[user.user_id]
+            completion.profile_description = user.bio is not None
+            completion.profile_name = user.name is not None
+            completion.profile_picture = user.profile_picture is not None or user.profile_picture_sizes is not None
+            completion.profile_cover_photo = user.cover_photo is not None or user.cover_photo_sizes is not None
+
+    def _handle_reposts(self, session, partial_completions):
+        user_ids = list(map(lambda x: x.user_id, partial_completions))
+        reposts = session.query(Repost).filter(
+            Repost.is_current == True,
+            Repost.user_id.in_(user_ids),
+            Repost.is_delete == False
+        ).all()
+        reposts_counter = Counter(map(lambda x: x.user_id, reposts))
+
+        for completion in partial_completions:
+            completion.reposts_complete = reposts_counter[completion.user_id] >= REPOST_THRESHOLD
+
+    def _handle_follow(self, session, partial_completions):
+        user_ids = list(map(lambda x: x.user_id, partial_completions))
+        follows = session.query(Follow).filter(
+            Follow.is_current == True,
+            Follow.follower_user_id.in_(user_ids),
+            Follow.is_delete == False
+        ).all()
+        follows_counter = Counter(map(lambda x: x.follower_user_id, follows))
+        for completion in partial_completions:
+            completion.follows_complete = follows_counter[completion.user_id] >= FOLLOW_THRESHOLD
+
+    def _handle_favorite(self, session, partial_completions):
+        user_ids = list(map(lambda x: x.user_id, partial_completions))
+        favorites = session.query(Save).filter(
+            Save.is_current == True,
+            Save.user_id.in_(user_ids),
+            Save.is_delete == False
+        ).all()
+        follows_counter = Counter(map(lambda x: x.user_id, favorites))
+        for completion in partial_completions:
+            completion.favorites_complete = follows_counter[completion.user_id] >= FAVORITES_THRESHOLD
+
+    def _get_steps_complete(self, partial_challenge):
+        return (
+            partial_challenge.profile_description +
+            partial_challenge.profile_name + 
+            partial_challenge.profile_picture +
+            partial_challenge.profile_cover_photo +
+            partial_challenge.follows_complete +
+            partial_challenge.favorites_complete +
+            partial_challenge.reposts_complete
+        )
+
+profile_challenge_manager = ChallengeManager('profile_completion', ProfileChallengeUpdater())
+
+# Accessors
+def get_profile_completion_challenges(session, user_ids):
+    return session.query(ProfileCompletionChallenge).filter(ProfileCompletionChallenge.user_id.in_(user_ids)).all()
+
+def get_users(session, user_ids):
+    return session.query(User).filter(User.user_id.in_(user_ids)).all()
```

Comment:
i think we don't actually need the entire `User` object here right? it's just user_id, bio, name, profile_picture, cover_photo - might be worth simplifying if we are pulling a ton of Users

Reply:
good optimization, i'll do that